File: basic/src/main/java/com/yuanshenbin/basic/log/LogAbstract.java
package com.yuanshenbin.basic.log;

public abstract class LogAbstract {
    
    public static final String TAG="ysb";

    abstract void i(String tag, String msg);

    abstract void i(String msg);

    abstract void e(String tag, String msg);

    abstract void e(String msg);

    abstract void d(String tag, String msg);

    abstract void d(String msg);

    abstract void v(String tag, String msg);

    abstract void v(String msg);

    abstract void w(String tag, String msg);

    abstract void w(String msg);

}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/log/LogAbstract.kt
package com.yuanshenbin.basic.log

/**
 * author : yuanshenbin
 * time   : 2023/3/9
 * desc   :
 */
abstract class LogAbstract {
    abstract fun i(tag: String?, msg: String?)
    abstract fun i(msg: String?)
    abstract fun e(tag: String?, msg: String?)
    abstract fun e(msg: String?)
    abstract fun d(tag: String?, msg: String?)
    abstract fun d(msg: String?)
    abstract fun v(tag: String?, msg: String?)
    abstract fun v(msg: String?)
    abstract fun w(tag: String?, msg: String?)
    abstract fun w(msg: String?)
    val tag: String
        get() = "ysb"
}